src/Microsoft.Azure.SignalR.Common/ServiceMessages/ServicePingMessageExtension.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.SignalR
+{
+    using ServicePingMessage = Microsoft.Azure.SignalR.Protocol.PingMessage;
+
+    public static class ServicePingMessageExtension
+    {
+        public static bool TryGetServiceStatusPingMessage(this ServicePingMessage message, out ServiceStatusPingMessage serviceStatus)
+        {
+            if (TryGetValue(message, ServiceStatusPingMessage.Key, out var value))
+            {
+                serviceStatus = new ServiceStatusPingMessage(value);
+                return true;
+            }
+
+            serviceStatus = null;
+            return false;
+        }
+
+        private static bool TryGetValue(ServicePingMessage pingMessage, string key, out string value)
+        {
+            if (pingMessage == null)
+            {
+                value = null;
+                return false;
+            }
+
+            int index = 0;
+            while (index < pingMessage.Messages.Length - 1)
+            {
+                var item1 = pingMessage.Messages[index];
+                var item2 = pingMessage.Messages[index + 1];
+                if (item1 == key)
+                {
+                    value = item2;
+                    return true;
+                }
+
+                index += 2;
+            }
```

Comment:
would normal ping mixed together, then the order not work?

Reply:
all the ping follows the same pattern, we use key to distinguish different kind of pings

Reply:
get it.

---
In reply to: [297487067](https://github.com/Azure/azure-signalr/pull/576#discussion_r297487067) [](ancestors = 297487067)